components/brave_wallet/browser/keyring_service.cc

```diff
@@ -521,11 +521,12 @@ void KeyringService::RequestUnlock() {
 
 HDKeyring* KeyringService::ResumeKeyring(const std::string& keyring_id,
                                          const std::string& password) {
-  if (!CreateEncryptorForKeyring(password, keyring_id)) {
+  if (!encryptor_ && !CreateEncryptorForKeyring(password, keyring_id)) {
     return nullptr;
   }
 
-  const std::string mnemonic = GetMnemonicForKeyringImpl(keyring_id);
+  const std::string mnemonic =
+      GetMnemonicForKeyringImpl(mojom::kDefaultKeyringId);
```

Comment:
as discussed on Slack, we need to store different encrypted mnemonic per keyring because we have separate encryptor for each keyring even though we share the same plaintext mnemonic among keyrings.

Reply:
done